examples/dummysolver/main_dummysolver.cc

```diff
@@ -36,41 +36,41 @@ try {
     // - Name of solver
     // - Configuration file name
     // - Solver rank
-
     const std::string solverName =
         getParamFromGroup<std::string>("preCICE", "SolverName");
     const std::string preciceConfigFilename =
         getParamFromGroup<std::string>("preCICE", "ConfigFileName");
     const std::string meshName =
         getParamFromGroup<std::string>("preCICE", "MeshName");
+    const precice::string_view meshNameView(meshName);
 
-    auto &couplingInterface = Dumux::Precice::CouplingAdapter::getInstance();
-    couplingInterface.announceSolver(solverName, preciceConfigFilename,
-                                     mpiHelper.rank(), mpiHelper.size());
-
+    auto &couplingParticipant = Dumux::Precice::CouplingAdapter::getInstance();
+    couplingParticipant.announceSolver(solverName, preciceConfigFilename,
+                                       mpiHelper.rank(), mpiHelper.size());
     std::cout << "DUMMY (" << mpiHelper.rank()
               << "): Running solver dummy with preCICE config file \""
               << preciceConfigFilename << "\", participant name \""
               << solverName << "\", and mesh name \"" << meshName << "\".\n";
 
-    const int dimensions = couplingInterface.getDimensions();
+    const int dimensions = couplingParticipant.getMeshDimensions(meshNameView);
     assert(dimensions == 3);
-    const std::string scalarDataWriteName =
-        (solverName == "SolverOne") ? "scalarDataOne" : "scalarDataTwo";
-    const std::string scalarDataReadName =
-        (solverName == "SolverOne") ? "scalarDataTwo" : "scalarDataOne";
-
-    const std::string vectorDataWriteName =
-        (solverName == "SolverOne") ? "vectorDataOne" : "vectorDataTwo";
-    const std::string vectorDataReadName =
-        (solverName == "SolverOne") ? "vectorDataTwo" : "vectorDataOne";
+    const precice::string_view scalarDataWriteName = std::string(
+        (solverName == "SolverOne") ? "scalarDataOne" : "scalarDataTwo");
+    const precice::string_view scalarDataReadName = std::string(
+        (solverName == "SolverOne") ? "scalarDataTwo" : "scalarDataOne");
+    const precice::string_view vectorDataWriteName = std::string(
+        (solverName == "SolverOne") ? "vectorDataOne" : "vectorDataTwo");
+    const precice::string_view vectorDataReadName = std::string(
+        (solverName == "SolverOne") ? "vectorDataTwo" : "vectorDataOne");
 
     const int numberOfVertices = 3;
 
     std::vector<double> writeScalarData(numberOfVertices);
+    std::vector<double> readScalarData(numberOfVertices);
     std::vector<double> writeVectorData(numberOfVertices * dimensions);
-    std::vector<double> vertices(numberOfVertices * dimensions);
-    std::vector<int> preciceVertexIDs(numberOfVertices);
+    std::vector<double> readVectorData(numberOfVertices * dimensions);
```

Comment:
We should consider merging the adapter functions `writeScalarData` and `writeVectorData` into `writeData`, and also similarly have a `readData` instead of two functions.

Reply:
Yes, we actually only have two functions for reading and writing from/to preCICE like `writeData` and `readData`. 
In the dummy cases we have four quantities because we had distinguished between scalar and vector reading/writing with four functions in the previous version, and Alexander would like to test four of them. I retain them since I don't think it's a bad idea to test if the functions could work with data of all dimensions.